### PR TITLE
Rename example modules

### PR DESCRIPTION
**Pull request by tiziano88**
The `--module` flag was assuming that the modules are specified as
"backend,frontend", but the run_example script was collecting them in a
different order, which ended up mixing up the two modules needed for the
chat example.

This commit renames the modules so that it's clear in which order they
are going to be specified, and flips the expected order of the module
flag to reflect the fact that the frontend node is the first one to be
specified.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
I think this is potentially still vulnerable to the same problem, because the underlying issue is that I mistakenly assumed `find` sorts its output.  Would #434 work instead?

**Comment by tiziano88** [comment]
In my experience this does fix the issue, I thought the problem was that for the chat app the modules are called `chat.wasm` and `chat_backend.wasm`, and therefore even sorting them leaves them in this order (`.` < `_`). Or maybe there is a setting somewhere that causes the order to be different for non alpha characters? It seemed pretty deterministic on my machine. Or, maybe it's based on inode number or some other hidden variable that it's different across machines. I still think this PR clarifies module names and their expected order, and the flag reflects more clearly the intent of the modules are specified in this order.

**Comment by daviddrysdale** [comment]
Digging into manpages etc., it seems that:
 - `find` will (mostly) [order according to](https://serverfault.com/questions/181787/find-command-default-sorting-order) directory/file creation order, which will be consistent across runs on the same machine (and maybe even between machines, if Git unpacks things in a consistent order) but not guaranteed.
 - `sort` sorts [according to the value](https://www.gnu.org/software/coreutils/manual/html_node/sort-invocation.html#sort-invocation) of the `LC_COLLATE` locale setting. 

```bash
% export LC_COLLATE=C
% find examples/target/wasm32-unknown-unknown/release/chat*.wasm | sort
examples/target/wasm32-unknown-unknown/release/chat.wasm
examples/target/wasm32-unknown-unknown/release/chat_backend.wasm
% export LC_COLLATE=en_US.UTF-8
% find examples/target/wasm32-unknown-unknown/release/chat*.wasm | sort
examples/target/wasm32-unknown-unknown/release/chat_backend.wasm
examples/target/wasm32-unknown-unknown/release/chat.wasm
```

So yeah, probably best to have belt-and-braces and do both this PR and #434

File: examples/chat/client/chat.cc [hunk 1]
```diff
@@ -38,7 +38,7 @@ ABSL_FLAG(std::string, manager_address, "127.0.0.1:8888",
           "Address of the Oak Manager to connect to");
 // TODO: separate these out into distinct flags when test scripting is able to cope.
 ABSL_FLAG(std::vector<std::string>, module, std::vector<std::string>{},
-          "Files containing the compiled WebAssembly modules (as 'backend,frontend')");
+          "Files containing the compiled WebAssembly modules (as 'frontend,backend')");
```
Comment (daviddrysdale) [inline examples/chat/client/chat.cc]: I think the`abitest` client code needs an equivalent flip - #436

**Pull request merged**